src/core/lib/surface/call.cc

```diff
@@ -1755,7 +1755,7 @@ static grpc_call_error call_start_batch(grpc_call* call, const grpc_op* ops,
         if (op->data.send_status_from_server.status_details != nullptr) {
           call->send_extra_metadata[1].md = grpc_mdelem_from_slices(
               GRPC_MDSTR_GRPC_MESSAGE,
-              grpc_slice_ref_internal(
+              grpc_slice_copy(
```

Comment:
See the other comment I made for that... yeah there's other edge cases.

I think we've covered the cases right now, so I'd like to avoid making the change to slice_ref_internal, and instead design a C++ API that actually covers our usage ergonomically and start migrating code to use it. This would be one of the things that it needs to cover!